engine/pkg/meta/internal/sqlkv/sql_impl.go

```diff
@@ -0,0 +1,325 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlkv
+
+import (
+	"context"
+	"database/sql"
+	"sync"
+	"time"
+
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+
+	sqlkvModel "github.com/pingcap/tiflow/engine/pkg/meta/internal/sqlkv/model"
+	metaModel "github.com/pingcap/tiflow/engine/pkg/meta/model"
+	ormModel "github.com/pingcap/tiflow/engine/pkg/orm/model"
+	"github.com/pingcap/tiflow/pkg/errors"
+	cerrors "github.com/pingcap/tiflow/pkg/errors"
+)
+
+// TODO: call InitializeEpochModel before Initialize any KVClient
+// if we use sql metastore
+
+// sqlImpl is the mysql-compatible implement for KVClient
+type sqlImpl struct {
+	// gorm claims to be thread safe
+	db           *gorm.DB
```

Comment:
What is `db` used for in contrast to `tableScopeDB`?

Reply:
`db` is used in `GenEpoch` which has no table scope because logic epoch has it's own backend table.